fastapi_auth_apikey/dependencies.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2025 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from typing import Annotated
+
+from odoo.api import Environment
+
+from odoo.addons.base.models.res_partner import Partner
+from odoo.addons.fastapi.dependencies import fastapi_endpoint, odoo_env
+from odoo.addons.fastapi.models.fastapi_endpoint import FastapiEndpoint
+
+from fastapi import Depends, HTTPException, status
+from fastapi.security import APIKeyHeader
+
+
+def apikey_authenticated_partner_impl(
+    api_key: Annotated[
+        str,
+        Depends(
+            APIKeyHeader(
+                name="api-key",
+                description="Odoo default apikey",
+            )
+        ),
+    ],
+    env: Annotated[Environment, Depends(odoo_env)],
+    endpoint: Annotated[FastapiEndpoint, Depends(fastapi_endpoint)],
+) -> Partner:
+    uid = env["res.users.apikeys"]._check_credentials(
+        scope=endpoint.apikey_auth_scope, key=api_key
+    )
+    partner = env["res.users"].sudo().browse(uid).exists().partner_id
+    if not partner:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED, detail="Incorrect API Key"
+        )
+    if endpoint.user_id.id and endpoint.user_id.id != uid:
```

Comment:
@Tisho99 You link your endpoint to only one user? The `user_id` on the endpoint is the technical user running the process. The authentication is a way to authenticate a partner requesting the endpoint. I'm may be wrong when reading your code but this line will restrict the use of this endpoint to the partner linked to the user_id defined on the endpoint.

Reply:
@lmignon How do you see this approach? Do you think it would be better to use a checkbox to indicate which users are allowed?